scripts/priceBoundPSMDeployUSDC.ts

```diff
@@ -0,0 +1,168 @@
+import { getImpersonatedSigner } from '@test/helpers';
+
+/// ---------------------------------------------------------------- ///
+///                                                                  ///
+///                Price Bound Peg Stability Module                  ///
+///                                                                  ///
+/// ---------------------------------------------------------------- ///
+
+/// Description: This module is used to manage the stability of the peg.
+
+/// Steps:
+///  0 - Deploy USDC/VOLT PriceBoundPegStabilityModule
+///  1 - Pause Redemptions on the VOLT PriceBoundPegStabilityModule
+///  2 - Move VOLT into the PriceBoundPegStabilityModule to allow trades
+
+/// This contract needs no roles in the system, just funding to allow swaps
+
+/// ---------------------------------------------------------------- ///
+
+const {
+  CORE,
+  USDC,
+  VOLT,
+  PRICE_BOUND_PSM,
+  VOLT_FUSE_PCV_DEPOSIT,
+  MINT_FEE_BASIS_POINTS,
+  REDEEM_FEE_BASIS_POINTS,
+  PROTOCOL_MULTISIG_ADDRESS,
+  ORACLE_PASS_THROUGH_ADDRESS
+} = config;
+
+const usdcReservesThreshold = ethers.constants.MaxUint256; /// max uint value so that we can never allocate surplus on this PSM to the pcv deposit
+const mintLimitPerSecond = ethers.utils.parseEther('10000'); /// 10k Volt minted per second max, however this value is useless as this PSM will not be given the minter role
+const voltPSMBufferCap = ethers.utils.parseEther('10000000'); /// 10m Volt max can be minted at a time, however this value is useless as this PSM will not be given the minter role
+const initialUSDCPSMVoltAmount = ethers.utils.parseEther('1941000'); /// 1.941m Volt which equals $2m at current prices
+
+/// Oracle price gets scaled up by 1e12 to account for the differences in decimals of USDC and VOLT.
+/// USDC has 6 decimals while Volt has 12, thus creating a difference that has to be normalized
+const voltUSDCDecimalsNormalizer = 12;
+/// Need to scale up price of floor and ceiling by 1e12 to account for decimal normalizer that is factored into price
+const voltFloorPrice = '9000000000000000';
+const voltCeilingPrice = '10000000000000000';
+
+/// Deploy the USDC PriceBoundPSM
+const deploy = async () => {
+  const voltPSMFactory = await ethers.getContractFactory('PriceBoundPSM');
+
+  /// Deploy USDC Peg Stability Module
+  /// PSM will trade VOLT between 100 cents and 111 cents. If the Volt price exceeds 111 cents,
+  /// the floor price will have to be moved lower as the oracle price is inverted.
+  /// If price is outside of this band, the PSM will not allow trades
+  const voltPSM = await voltPSMFactory.deploy(
+    voltFloorPrice,
+    voltCeilingPrice,
+    {
+      coreAddress: CORE,
+      oracleAddress: ORACLE_PASS_THROUGH_ADDRESS, /// OPT
+      backupOracle: ethers.constants.AddressZero,
+      decimalsNormalizer: voltUSDCDecimalsNormalizer,
+      doInvert: true /// invert the price so that the Oracle and PSM works correctly
```

Comment:
Why exactly does this need to be inverted?

Reply:
The oracle returns the price of Volt in dollars. However, the PSM has volt as the native asset, and stablecoins as the external asset. The PSM was originally designed for the native asset to be stable, and to fetch the oracle price of the external asset. However, this PSM breaks that assumption and inverts the order, it fetches the native token price from an oracle, and assumes the external asset is stable.